Subject: Eliminate an autoconfig warning.
--- hdf-4.2.13/configure.ac	2017-06-24 22:00:04.000000000 +0100
+++ hdf/configure.ac	2019-01-22 11:38:09.165958293 +0000
@@ -33,7 +33,7 @@ AC_CONFIG_MACRO_DIR([m4])
 
 ## AM_INIT_AUTOMAKE takes a list of options that should be applied to
 ## every Makefile.am when automake is run.
-AM_INIT_AUTOMAKE([foreign])
+AM_INIT_AUTOMAKE([foreign subdir-objects])
 
 ## AM_MAINTAINER_MODE turns off "rebuild rules" that contain dependencies
 ## for Makefiles, configure, etc.  If AM_MAINTAINER_MODE
