--- a/HDF4options.cmake	2021-12-06 16:50:41.040077058 +0000
+++ b/HDF4options.cmake	2021-12-06 16:52:10.073330865 +0000
@@ -15,6 +15,7 @@
 ### ext libs on system
 set(ADD_BUILD_OPTIONS "${ADD_BUILD_OPTIONS} -DCMAKE_PREFIX_PATH:FILEPATH=${LIB_DIR}")
 set(ADD_BUILD_OPTIONS "${ADD_BUILD_OPTIONS} -DSZIP_LIBRARY:FILEPATH=${LIB_DIR}/lib/libsz.a -DSZIP_INCLUDE_DIR:PATH=${LIB_DIR}/include")
+set(ADD_BUILD_OPTIONS "${ADD_BUILD_OPTIONS} -DCMAKE_Fortran_FLAGS='-fallow-argument-mismatch'")
 
 ##############################################################################################
 ### disable test program builds
