From f173ecba7134c10824487edeeb8d28c86cfb64be Mon Sep 17 00:00:00 2001
From: Michael Sarahan <msarahan@gmail.com>
Date: Thu, 30 Nov 2017 22:11:20 -0600
Subject: [PATCH] fix automake files for linux compatibility

---
 configure.ac    | 29 +++++++++++++++++++++++++++++
 src/Makefile.am |  9 ++++++++-
 2 files changed, 37 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 921f90f..aed32b9 100755
--- a/configure.ac
+++ b/configure.ac
@@ -20,6 +20,35 @@ AC_DISABLE_SHARED
 AM_PROG_LIBTOOL
 
 
+# AC_CANONICAL_HOST is needed to access the 'host_os' variable
+AC_CANONICAL_HOST
+
+build_linux=no
+build_windows=no
+build_mac=no
+
+# Detect the target system
+case "${host_os}" in
+linux*)
+build_linux=yes
+;;
+cygwin*|mingw*)
+build_windows=yes
+;;
+darwin*)
+build_mac=yes
+;;
+*)
+AC_MSG_ERROR(["OS $host_os is not supported"])
+;;
+esac
+
+# Pass the conditionals to automake
+AM_CONDITIONAL([LINUX], [test "$build_linux" = "yes"])
+AM_CONDITIONAL([WINDOWS], [test "$build_windows" = "yes"])
+AM_CONDITIONAL([OSX], [test "$build_mac" = "yes"])
+
+
 dnl --------------------------------------------------
 dnl Set prefix default (install directory) to a directory
 dnl in the build area.
diff --git a/src/Makefile.am b/src/Makefile.am
index f87d373..65036b8 100755
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -6,7 +6,14 @@ include $(top_srcdir)/config/include.am
 INCLUDES=-I$(top_srcdir)/include/
 
 # Set LDFLAGS to alow the HDF-EOS library to use extern variables from HDF4
-LDFLAGS=-Wl,-single_module
+if LINUX
+LDFLAGS+= -shrext .so
+LIBS+= -lm
+endif
+
+if OSX
+LDFLAGS+= -Wl,-single_module
+endif
 
 # Build HDF-EOS2
 lib_LTLIBRARIES=libhdfeos.la
-- 
2.11.1

