Subject: [PATCH] fix automake files for linux compatibility
--- a/configure.ac	2021-05-18 12:55:16.000000000 +0100
+++ b/configure.ac	2021-12-02 14:41:17.970040499 +0000
@@ -16,6 +16,35 @@
 AM_MAINTAINER_MODE
 
 
+# AC_CANONICAL_HOST is needed to access the 'host_os' variable
+AC_CANONICAL_HOST
+
+build_linux=no
+build_windows=no
+build_mac=no
+
+# Detect the target system
+case "${host_os}" in
+linux*)
+build_linux=yes
+;;
+cygwin*|mingw*)
+build_windows=yes
+;;
+darwin*)
+build_mac=yes
+;;
+*)
+AC_MSG_ERROR(["OS $host_os is not supported"])
+;;
+esac
+
+# Pass the conditionals to automake
+AM_CONDITIONAL([LINUX], [test "$build_linux" = "yes"])
+AM_CONDITIONAL([WINDOWS], [test "$build_windows" = "yes"])
+AM_CONDITIONAL([OSX], [test "$build_mac" = "yes"])
+
+
 ## Note we disable shared libraries otherwise gfortran fails linking against the resulting
 ## libX.la files. With shared libraries disables, libtool links gfortran against the static
 ## library libX.a. If fortran is disabled, this can be overridden using --enabled-shared
@@ -91,10 +120,6 @@
 # we only need to find the libary, not the header files.
 
 HDF_CHECK_LIB([jpeg],[jpeg],[jpeg_set_defaults],[,/usr/local/lib,/opt/local/lib,/usr/lib64])
-AS_IF(
-    [test -z "$HAVE_LIBJPEG"],
-    [AC_MSG_ERROR(unable to find libjpeg)]
-)
 
 
 
@@ -106,10 +131,6 @@
 # we only need to find the libary, not the header files.
 
 HDF_CHECK_LIB([zlib],[z],[compress2],[,/usr/local/lib,/opt/local/lib,/usr/lib64])
-AS_IF(
-    [test -z "$HAVE_LIBZ"],
-    [AC_MSG_ERROR(unable to find libz)]
-)
 
 
 
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -19,7 +19,14 @@
 
 # Set LDFLAGS to alow the HDF-EOS library to use extern variables from HDF4
-#LDFLAGS=-Wl,-single_module
+if LINUX
+LDFLAGS+= -shrext .so
+LIBS+= -lm
+endif
+
+if OSX
+LDFLAGS+= -Wl,-single_module
+endif
 
 
