Subject: Eliminate an autoconfig warning.
--- hdfeos-2.20/configure.ac	2009-05-19 17:12:22.000000000 +0100
+++ hdfeos/configure.ac         2019-01-22 12:35:55.513755431 +0000
@@ -11,7 +11,7 @@ AC_CONFIG_HEADER([include/HE2_config.h])
 
 AC_CONFIG_AUX_DIR([config])
 
-AM_INIT_AUTOMAKE([foreign])
+AM_INIT_AUTOMAKE([foreign subdir-objects])
 AM_MAINTAINER_MODE
 
 # Disable shared libraries (for now)
