Subject: Incorrect line continuation in test scripts.
diff -rupN hdfeos2_1549881068987/work/samples/writefields.f hdfeos2/work/samples/writefields.f
--- hdfeos2_1549881068987/samples/writefields.f	2017-03-08 14:10:32.000000000 +0000
+++ hdfeos27/samples/writefields.f	2019-02-11 10:44:08.824490210 +0000
@@ -32,7 +32,7 @@ c     Open the HDF swath file, "SwathFil
 c 
 
 	swfid = swopen("SwathFile_created_with_hadeos_sample_file_write"//
-     1r_of_HDFEOS2_version_219_or_higher_release.hdf", DFACC_RDWR)
+     1"r_of_HDFEOS2_version_219_or_higher_release.hdf", DFACC_RDWR)
 
 
 	if (swfid .NE. -1) then
