Subject: Fix use of uninitialised pointer.
diff -upN hdfeos/src/SWapi.c hdfeos2/src/SWapi.c
--- a/src/SWapi.c	2019-02-11 13:01:30.721040054 +0000
+++ b/src/SWapi.c	2019-02-11 12:55:20.371945678 +0000
@@ -13976,7 +13976,7 @@ SWllmapedidxinfo(int32 swathID, const ch
   char            dimname0[HDFE_DIMBUFSIZE],dimname1[HDFE_DIMBUFSIZE];
   char            geodim[256];   /* Geolocation field dimension list */
   char            geodimbuf[256];/* Geolocation field dimension list */
-  int32           nDim;	         /* Number of dimensions            */
+  int32           nDim, nDim2;   /* Number of dimensions            */
   int32           slen[64],slen2[64];       /* String length array  */
   char           *ptr[64],*ptr2[64];	    /* String pointer array */
   int32           dumdims[8];	     /* Dimensions from SWfieldinfo */
@@ -14019,7 +14019,7 @@ SWllmapedidxinfo(int32 swathID, const ch
       l_index[iDim] = 0;
     }
 
-  for(iDim=0; iDim<8; iDim++)
+  for(iDim=0; iDim<nDim; iDim++)
     {
 
       if(ptr[iDim] == NULL) break;
@@ -14034,12 +14034,12 @@ SWllmapedidxinfo(int32 swathID, const ch
 
       /* Parse Data Field Dimlist & find mapping */
       /* --------------------------------------- */
-      nDim = EHparsestr(dimlist, ',', ptr2, slen2);
+      nDim2 = EHparsestr(dimlist, ',', ptr2, slen2);
       
   
       /* Loop through all dimensions and search for mapping */
       /* -------------------------------------------------- */
-      for (i = 0; i < nDim; i++)
+      for (i = 0; i < nDim2; i++)
 	{
 	  memcpy(utlbuf, ptr2[i], slen2[i]);
 	  utlbuf[slen2[i]] = 0;
@@ -14087,7 +14087,7 @@ SWllmapedidxinfo(int32 swathID, const ch
  	  
 	  /* Loop through all dimensions and search for mapping */
 	  /* -------------------------------------------------- */
-	  for (i = 0; i < nDim; i++)
+	  for (i = 0; i < nDim2; i++)
 	    {
 	      memcpy(utlbuf, ptr2[i], slen2[i]);
 	      utlbuf[slen2[i]] = 0;
