Subject: Use curl rather than wget.
--- netcdf-c-4.6.2/hdf4_test/run_get_hdf4_files.sh	(date 1542880547000)
+++ netcdf-c-4.6.2/hdf4_test/run_get_hdf4_files.sh	(date 1542880568000)
@@ -18,10 +18,10 @@
    for try in 1 2 3 4 ; do # try 4 times

      # signal success/failure
-     if wget -c $Q --passive-ftp $FTPFILE ; then
+     if curl -O $Q $FTPFILE ; then
        return 0 # got it
      fi
-     echo "wget failed: try $try"
+     echo "curl failed: try $try"
      sleep 5 # seconds
    done
    return 1 # did not get it
