--- /libnetcdf7.4.7/libhdf5/hdf5var.c	2020-03-27 15:33:36.000000000 +0000
+++ /libnetcdf/libhdf5/hdf5var.c	2020-11-18 12:19:29.105648604 +0000
@@ -2290,7 +2290,7 @@
     float real_preemption = CHUNK_CACHE_PREEMPTION;
 
     LOG((1, "%s: ncid 0x%x varid %d size %d nelems %d preemption %d",
-	 __func__, ncid, varid, size, nelems, preemptions));
+	 __func__, ncid, varid, size, nelems, preemption));
     
     if (size >= 0)
         real_size = ((size_t) size) * MEGABYTE;