--- a/make.inc	2022-07-06 13:46:14.834354973 +0100
+++ b/make.inc	2019-10-31 13:53:18.271095000 +0000
@@ -1,12 +1,12 @@
 # C compiler and C compiler flags
 
-#CC      = $(CC)
-CCFLAGS = $(CFLAGS) -O2
+CC      = $(basename $(GCC))
+CCFLAGS = $(CFLAGS)
 
 # Fortran compiler and Fortan compiler flags (required for the Fortran
 # interface)
-F90      = $(FC)
-F90FLAGS = $(FFLAGS) -O2
+F90      = $(basename $(GFORTRAN))
+F90FLAGS = $(FFLAGS)
 
 LINKS = -lm
 
