From 67a20c4a65a7e729a03e8a5f3594cbf8abf625da Mon Sep 17 00:00:00 2001
From: Daniel Philipp <51949198+danielphilipp@users.noreply.github.com>
Date: Tue, 22 Nov 2022 12:33:22 +0100
Subject: [PATCH 1/6] Improvements to Rayleigh scattering handling

---
 pre_processing/rttov_driver.F90 | 46 ++++++++++++++++++++++++---------
 1 file changed, 34 insertions(+), 12 deletions(-)

diff --git a/pre_processing/rttov_driver.F90 b/pre_processing/rttov_driver.F90
index 9c39c306..47a6493d 100644
--- a/pre_processing/rttov_driver.F90
+++ b/pre_processing/rttov_driver.F90
@@ -146,6 +146,11 @@
 ! 2019/08/15, SP: Add check for good pixels, meaning we don't run RTTOV on
 !                 those we don't care about. This gives a big speedup to
 !                 processing instruments that cross the dateline.
+! 2022/11/22, DP: Default behaviour switches RTTOV Rayleigh scattering off
+!                 and thus ORAC does not need to call remove_rayleigh(). 
+!                 Additionally pass calcrefl(:)=.false. and 
+!                 reflectance%refl_in=0. vectors to RTTOV to overcome the RTTOV 
+!                 error with some compilers when opts%rt_ir%addsolar=.true..
 !
 ! Bugs:
 ! - BRDF not yet implemented here, so RTTOV internal calculation used.
@@ -235,6 +240,8 @@ subroutine rttov_driver(coef_path, emiss_path, granule, preproc_dims, &
    logical(kind=jplm),      allocatable :: calcemis(:)
    type(rttov_emissivity),  allocatable :: emissivity(:)
    real(kind=jprb),         allocatable :: emis_data(:)
+   logical(kind=jplm),      allocatable :: calcrefl(:)
+   type(rttov_reflectance), allocatable :: reflectance(:)
    type(rttov_transmission)             :: transmission
    type(rttov_radiance)                 :: radiance
    type(rttov_radiance2)                :: radiance2
@@ -436,10 +443,11 @@ subroutine rttov_driver(coef_path, emiss_path, granule, preproc_dims, &
    ! regression limits (NB by doing this the extrapolated values outside
    ! the regression limits will be reset to the limits: it will not result
    ! in unphysical extrapolated profile values being used)
-   opts % rt_all % use_q2m   = .false. ! Do not use surface humidity
-   opts % rt_all % addrefrac = .true.  ! Include refraction in path calc
-   opts % rt_ir % addsolar   = .true.  ! Do not include reflected solar
-   opts % rt_all % ozone_data = .true.  ! Include ozone profile
+   opts % rt_all % use_q2m                = .false. ! Do not use surface humidity
+   opts % rt_all % addrefrac              = .true.  ! Include refraction in path calc
+   opts % rt_ir % addsolar                = .true.  ! Do not include reflected solar
+   opts % rt_ir % rayleigh_max_wavelength = 0.      ! Do not run Rayleigh in RTTOV
+   opts % rt_all % ozone_data             = .true.  ! Include ozone profile
    if (pre_opts%do_co2) then
       opts % rt_all % co2_data   = .true.  ! Include CO2 profile
    else
@@ -736,6 +744,15 @@ subroutine rttov_driver(coef_path, emiss_path, granule, preproc_dims, &
          allocate(emissivity(nchan))
          allocate(emis_data(nchan))
          allocate(calcemis(nchan))
+         allocate(reflectance(nchan))
+         allocate(calcrefl(nchan))
+         
+         ! These arrays are needed when running with opts%rt_ir%addsolar
+         ! as RTTOV throws and error with some compilers. Set reflectance
+         ! vector to 0. as the surface reflecatance is explicitly handled
+         ! by the foreward model.
+         reflectance(:)%refl_in = 0.
+         calcrefl(:) = .false.
 
          chanprof%prof = 1
          do j = 1, nchan
@@ -819,11 +836,11 @@ subroutine rttov_driver(coef_path, emiss_path, granule, preproc_dims, &
 #ifdef INCLUDE_RTTOV_OPENMP
                      call rttov_parallel_direct(stat, chanprof, opts, &
                           profiles(count:count), coefs, transmission, radiance, &
-                          radiance2, calcemis, emissivity, traj=traj)
+                          radiance2, calcemis, emissivity, calcrefl, reflectance, traj=traj)
 #else
                      call rttov_direct(stat, chanprof, opts, &
                           profiles(count:count), coefs, transmission, radiance, &
-                          radiance2, calcemis, emissivity, traj=traj)
+                          radiance2, calcemis, emissivity, calcrefl, reflectance, traj=traj)
 #endif
 
                      if (stat /= errorstatus_success) then
@@ -831,11 +848,14 @@ subroutine rttov_driver(coef_path, emiss_path, granule, preproc_dims, &
                         stop error_stop_code
                      end if
 
-                     ! Remove the Rayleigh component from the RTTOV tranmittances.
+                     ! Remove the Rayleigh component from the RTTOV tranmittances only if RTTOV
+                     ! ran with Rayleigh scattering switched on.
                      if (i_coef == 2) then
-                        call remove_rayleigh(nchan, nlevels, dummy_sreal_1dveca, &
-                             profiles(count)%zenangle, profiles(count)%p, &
-                             transmission%tau_levels, transmission%tau_total)
+                        if (opts%rt_ir%rayleigh_max_wavelength > dither_more) then
+                           call remove_rayleigh(nchan, nlevels, dummy_sreal_1dveca, &
+                                profiles(count)%zenangle, profiles(count)%p, &
+                                transmission%tau_levels, transmission%tau_total)
+                         end if
                      end if
 
                   end if
@@ -883,11 +903,11 @@ subroutine rttov_driver(coef_path, emiss_path, granule, preproc_dims, &
 #ifdef INCLUDE_RTTOV_OPENMP
                      call rttov_parallel_direct(stat, chanprof, opts, &
                           profiles(count:count), coefs, transmission, radiance, &
-                          radiance2, calcemis, emissivity, traj=traj)
+                          radiance2, calcemis, emissivity, calcrefl, reflectance, traj=traj)
 #else
                      call rttov_direct(stat, chanprof, opts, &
                           profiles(count:count), coefs, transmission, radiance, &
-                          radiance2, calcemis, emissivity, traj=traj)
+                          radiance2, calcemis, emissivity, calcrefl, reflectance, traj=traj)
 #endif
 
                      ! Save into the appropriate arrays
@@ -917,6 +937,8 @@ subroutine rttov_driver(coef_path, emiss_path, granule, preproc_dims, &
          deallocate(emis_data)
          deallocate(calcemis)
          deallocate(chan_pos)
+         deallocate(calcrefl)
+         deallocate(reflectance)
       end do !coef loop
    end do  !view loop
 

From 0157c7bcd3fdf2e071439ed9c4ae314ece5bd922 Mon Sep 17 00:00:00 2001
From: Daniel Philipp <51949198+danielphilipp@users.noreply.github.com>
Date: Tue, 22 Nov 2022 12:52:51 +0100
Subject: [PATCH 2/6] Update rttov_driver.F90

---
 pre_processing/rttov_driver.F90 | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pre_processing/rttov_driver.F90 b/pre_processing/rttov_driver.F90
index 47a6493d..323dea95 100644
--- a/pre_processing/rttov_driver.F90
+++ b/pre_processing/rttov_driver.F90
@@ -851,7 +851,7 @@ subroutine rttov_driver(coef_path, emiss_path, granule, preproc_dims, &
                      ! Remove the Rayleigh component from the RTTOV tranmittances only if RTTOV
                      ! ran with Rayleigh scattering switched on.
                      if (i_coef == 2) then
-                        if (opts%rt_ir%rayleigh_max_wavelength > dither_more) then
+                        if (opts%rt_ir%rayleigh_max_wavelength > dither) then
                            call remove_rayleigh(nchan, nlevels, dummy_sreal_1dveca, &
                                 profiles(count)%zenangle, profiles(count)%p, &
                                 transmission%tau_levels, transmission%tau_total)

From ee3fdbae2d28f0de8adf65b467448cf29e33e190 Mon Sep 17 00:00:00 2001
From: Daniel Philipp <51949198+danielphilipp@users.noreply.github.com>
Date: Tue, 22 Nov 2022 13:03:38 +0100
Subject: [PATCH 3/6] Minor formatting

---
 pre_processing/rttov_driver.F90 | 12 ++++++++----
 1 file changed, 8 insertions(+), 4 deletions(-)

diff --git a/pre_processing/rttov_driver.F90 b/pre_processing/rttov_driver.F90
index 323dea95..cd8d87df 100644
--- a/pre_processing/rttov_driver.F90
+++ b/pre_processing/rttov_driver.F90
@@ -836,11 +836,13 @@ subroutine rttov_driver(coef_path, emiss_path, granule, preproc_dims, &
 #ifdef INCLUDE_RTTOV_OPENMP
                      call rttov_parallel_direct(stat, chanprof, opts, &
                           profiles(count:count), coefs, transmission, radiance, &
-                          radiance2, calcemis, emissivity, calcrefl, reflectance, traj=traj)
+                          radiance2, calcemis, emissivity, calcrefl, reflectance, &
+                          traj=traj)
 #else
                      call rttov_direct(stat, chanprof, opts, &
                           profiles(count:count), coefs, transmission, radiance, &
-                          radiance2, calcemis, emissivity, calcrefl, reflectance, traj=traj)
+                          radiance2, calcemis, emissivity, calcrefl, reflectance, &
+                          traj=traj)
 #endif
 
                      if (stat /= errorstatus_success) then
@@ -903,11 +905,13 @@ subroutine rttov_driver(coef_path, emiss_path, granule, preproc_dims, &
 #ifdef INCLUDE_RTTOV_OPENMP
                      call rttov_parallel_direct(stat, chanprof, opts, &
                           profiles(count:count), coefs, transmission, radiance, &
-                          radiance2, calcemis, emissivity, calcrefl, reflectance, traj=traj)
+                          radiance2, calcemis, emissivity, calcrefl, reflectance, &
+                          traj=traj)
 #else
                      call rttov_direct(stat, chanprof, opts, &
                           profiles(count:count), coefs, transmission, radiance, &
-                          radiance2, calcemis, emissivity, calcrefl, reflectance, traj=traj)
+                          radiance2, calcemis, emissivity, calcrefl, reflectance, &
+                          traj=traj)
 #endif
 
                      ! Save into the appropriate arrays

From 2e6f19dae3bd7386e4766e690e9ebc81dffb7a50 Mon Sep 17 00:00:00 2001
From: Daniel Philipp <51949198+danielphilipp@users.noreply.github.com>
Date: Mon, 28 Nov 2022 08:31:21 +0100
Subject: [PATCH 4/6] Re-introduce old Rayleigh scattering treatment

---
 pre_processing/rttov_driver.F90 | 17 +++++++----------
 1 file changed, 7 insertions(+), 10 deletions(-)

diff --git a/pre_processing/rttov_driver.F90 b/pre_processing/rttov_driver.F90
index cd8d87df..ae6a5b63 100644
--- a/pre_processing/rttov_driver.F90
+++ b/pre_processing/rttov_driver.F90
@@ -443,11 +443,10 @@ subroutine rttov_driver(coef_path, emiss_path, granule, preproc_dims, &
    ! regression limits (NB by doing this the extrapolated values outside
    ! the regression limits will be reset to the limits: it will not result
    ! in unphysical extrapolated profile values being used)
-   opts % rt_all % use_q2m                = .false. ! Do not use surface humidity
-   opts % rt_all % addrefrac              = .true.  ! Include refraction in path calc
-   opts % rt_ir % addsolar                = .true.  ! Do not include reflected solar
-   opts % rt_ir % rayleigh_max_wavelength = 0.      ! Do not run Rayleigh in RTTOV
-   opts % rt_all % ozone_data             = .true.  ! Include ozone profile
+   opts % rt_all % use_q2m         = .false. ! Do not use surface humidity
+   opts % rt_all % addrefrac       = .true.  ! Include refraction in path calc
+   opts % rt_ir % addsolar         = .true.  ! Do not include reflected solar
+   opts % rt_all % ozone_data      = .true.  ! Include ozone profile
    if (pre_opts%do_co2) then
       opts % rt_all % co2_data   = .true.  ! Include CO2 profile
    else
@@ -853,11 +852,9 @@ subroutine rttov_driver(coef_path, emiss_path, granule, preproc_dims, &
                      ! Remove the Rayleigh component from the RTTOV tranmittances only if RTTOV
                      ! ran with Rayleigh scattering switched on.
                      if (i_coef == 2) then
-                        if (opts%rt_ir%rayleigh_max_wavelength > dither) then
-                           call remove_rayleigh(nchan, nlevels, dummy_sreal_1dveca, &
-                                profiles(count)%zenangle, profiles(count)%p, &
-                                transmission%tau_levels, transmission%tau_total)
-                         end if
+                        call remove_rayleigh(nchan, nlevels, dummy_sreal_1dveca, &
+                             profiles(count)%zenangle, profiles(count)%p, &
+                             transmission%tau_levels, transmission%tau_total)
                      end if
 
                   end if

From 01a9f9f84b08614056ef9161861a858fa542968e Mon Sep 17 00:00:00 2001
From: Daniel Philipp <51949198+danielphilipp@users.noreply.github.com>
Date: Mon, 28 Nov 2022 11:53:16 +0100
Subject: [PATCH 5/6] Update rttov_driver.F90

---
 pre_processing/rttov_driver.F90 | 17 ++++++++++-------
 1 file changed, 10 insertions(+), 7 deletions(-)

diff --git a/pre_processing/rttov_driver.F90 b/pre_processing/rttov_driver.F90
index ae6a5b63..133cf30a 100644
--- a/pre_processing/rttov_driver.F90
+++ b/pre_processing/rttov_driver.F90
@@ -443,10 +443,11 @@ subroutine rttov_driver(coef_path, emiss_path, granule, preproc_dims, &
    ! regression limits (NB by doing this the extrapolated values outside
    ! the regression limits will be reset to the limits: it will not result
    ! in unphysical extrapolated profile values being used)
-   opts % rt_all % use_q2m         = .false. ! Do not use surface humidity
-   opts % rt_all % addrefrac       = .true.  ! Include refraction in path calc
-   opts % rt_ir % addsolar         = .true.  ! Do not include reflected solar
-   opts % rt_all % ozone_data      = .true.  ! Include ozone profile
+   opts % rt_all % use_q2m            = .false. ! Do not use surface humidity
+   opts % rt_all % addrefrac          = .true.  ! Include refraction in path calc
+   opts % rt_ir % addsolar            = .true.  ! Do not include reflected solar
+   opts%rt_ir%rayleigh_max_wavelength = 0. ! Do not run RTTOV Rayleigh scattering
+   opts % rt_all % ozone_data         = .true.  ! Include ozone profile
    if (pre_opts%do_co2) then
       opts % rt_all % co2_data   = .true.  ! Include CO2 profile
    else
@@ -852,9 +853,11 @@ subroutine rttov_driver(coef_path, emiss_path, granule, preproc_dims, &
                      ! Remove the Rayleigh component from the RTTOV tranmittances only if RTTOV
                      ! ran with Rayleigh scattering switched on.
                      if (i_coef == 2) then
-                        call remove_rayleigh(nchan, nlevels, dummy_sreal_1dveca, &
-                             profiles(count)%zenangle, profiles(count)%p, &
-                             transmission%tau_levels, transmission%tau_total)
+                        if (opts%rt_ir%rayleigh_max_wavelength > dither) then
+                           call remove_rayleigh(nchan, nlevels, dummy_sreal_1dveca, &
+                                profiles(count)%zenangle, profiles(count)%p, &
+                                transmission%tau_levels, transmission%tau_total)
+                        end if
                      end if
 
                   end if

From ae952462a0865bd2ab06cc5219b839930fba0005 Mon Sep 17 00:00:00 2001
From: Daniel Philipp <51949198+danielphilipp@users.noreply.github.com>
Date: Mon, 28 Nov 2022 11:53:53 +0100
Subject: [PATCH 6/6] Some formatting

---
 pre_processing/rttov_driver.F90 | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pre_processing/rttov_driver.F90 b/pre_processing/rttov_driver.F90
index 133cf30a..be29ec8a 100644
--- a/pre_processing/rttov_driver.F90
+++ b/pre_processing/rttov_driver.F90
@@ -446,7 +446,7 @@ subroutine rttov_driver(coef_path, emiss_path, granule, preproc_dims, &
    opts % rt_all % use_q2m            = .false. ! Do not use surface humidity
    opts % rt_all % addrefrac          = .true.  ! Include refraction in path calc
    opts % rt_ir % addsolar            = .true.  ! Do not include reflected solar
-   opts%rt_ir%rayleigh_max_wavelength = 0. ! Do not run RTTOV Rayleigh scattering
+   opts%rt_ir%rayleigh_max_wavelength = 0.      ! Do not run RTTOV Rayleigh scattering
    opts % rt_all % ozone_data         = .true.  ! Include ozone profile
    if (pre_opts%do_co2) then
       opts % rt_all % co2_data   = .true.  ! Include CO2 profile
