Subject: Adapt config files to the anaconda environment without openmp.
--- a/config/arch.conda.inc	2022-08-12 13:18:54.203033707 +0100
+++ b/config/arch.conda.inc	2022-08-12 16:30:41.874308585 +0100
@@ -21,11 +21,11 @@ F77FLAGS  = -O3 -cpp -ffree-line-length-
 # Set Fortran 90 compiler flags
 FFLAGS    = -O3 -cpp -ffree-line-length-0 -g -fcheck=bounds -fallow-argument-mismatch
 # Uncomment if OpenMP support is desired. Highly recommented.
-FFLAGS   += -fopenmp
+#FFLAGS   += -fopenmp
 # Uncomment if extra warnings are required.  Take out -Werror if that annoys you.
 #FFLAGS   += -Wall -Werror -Wno-conversion -Wno-uninitialized -Wno-unused-dummy-argument -Wtabs
 
-LFLAGS = -fopenmp
+LFLAGS =
 
 # Set C compiler  flags
 CFLAGS = -O3 -g
--- a/config/lib.conda.inc	2022-08-12 13:18:54.203033707 +0100
+++ b/config/lib.conda.inc	2022-08-12 16:31:17.946772047 +0100
@@ -1,8 +1,8 @@
 # Base directory for all libraries and include files taken from the
-# environmental variable CONDA_PREFIX.
+# environmental variable PREFIX.
 
 # Set up libraries and includes
-LIBS = -L$(CONDA_PREFIX)/lib \
+LIBS = -L$(PREFIX)/lib \
        -lemosR64 -lemos -lfftw3 \
        -lhdfeos -lGctp \
        -leccodes_f90 -leccodes \
@@ -16,17 +16,17 @@
        -ljpeg -lm -lz -lstdc++ -lblas -llapack
 
 INC = -I./ \
-      -I$(CONDA_PREFIX)/include \
-      -I$(CONDA_PREFIX)/mod
+      -I$(PREFIX)/include \
+      -I$(PREFIX)/mod
 
 CINC = -I./ \
-       -I$(CONDA_PREFIX)/include
+       -I$(PREFIX)/include
 
 
 # Configuration options
 
 # Uncomment if you want to enable OpenMP for RTTOV computations.
-INC  += -DINCLUDE_RTTOV_OPENMP
+#INC  += -DINCLUDE_RTTOV_OPENMP
 
 # Uncomment if ATSR support is desired.
 LIBS += -lepr_api
--- a/common/Makefile	2022-07-08 15:35:55.173943901 +0100
+++ b/common/Makefile	2023-02-13 11:01:37.296430262 +0000
@@ -59,7 +59,7 @@
 
 # Main library build
 liborac_common.a: $(COMMON_OBJ) Makefile
-	ar -rs liborac_common.a $(OBJS)/*.o
+	$(AR) -rs liborac_common.a $(OBJS)/*.o
 
 # Small program to report the full version number of RTTOV used
 rttov_version: $(OBJS)/rttov_version.o
--- a/derived_products/broadband_fluxes/Makefile	2020-04-21 12:21:34.000000000 +0100
+++ b/derived_products/broadband_fluxes/Makefile	2023-02-13 11:03:10.694189067 +0000
@@ -80,7 +80,7 @@
 	$(F90) -o process_broadband_fluxes $(LFLAGS) $(ORAC_OBJ_MAIN) libprocess_broadband_fluxes.a $(ORAC_COMMON)/liborac_common.a -I./$(OBJS) $(LIBS)
 
 libprocess_broadband_fluxes.a: $(ORAC_OBJ) Makefile
-	ar -rs libprocess_broadband_fluxes.a $(ORAC_OBJ)
+	$(AR) -rs libprocess_broadband_fluxes.a $(ORAC_OBJ)
 
 # Rebuild dependencies housed in dependencies.inc
 depend:
--- a/post_processing/Makefile	2020-04-21 12:21:30.000000000 +0100
+++ b/post_processing/Makefile	2023-02-13 11:03:59.955163894 +0000
@@ -51,7 +51,7 @@
 
 # Main library build
 libpostproc.a: $(POSTPROC_OBJ) Makefile
-	ar -rs libpostproc.a $(POSTPROC_OBJ)
+	$(AR) -rs libpostproc.a $(POSTPROC_OBJ)
 
 # Rebuild dependencies housed in dependencies.inc
 depend:
--- a/pre_processing/Makefile	2022-07-05 14:11:19.702955751 +0100
+++ b/pre_processing/Makefile	2023-02-13 11:05:06.221520069 +0000
@@ -125,7 +125,7 @@ orac_preproc: $(PREPROC_OBJ_MAIN) libpre
 
 # Main library build
 libpreproc.a: $(PREPROC_OBJ) Makefile
-	ar -rs libpreproc.a $(PREPROC_OBJ)
+	$(AR) -rs libpreproc.a $(PREPROC_OBJ)
 
 # Rebuild dependencies housed in dependencies.inc
 depend:
--- a/src/Makefile	2021-09-14 17:34:55.000000000 +0100
+++ b/src/Makefile	2023-02-13 11:05:45.375343222 +0000
@@ -233,7 +233,7 @@ orac: $(ORAC_OBJ_MAIN) liborac.a $(ORAC_
 
 # Main library build
 liborac.a: $(ORAC_OBJ) $(PARSER_OBJ) Makefile
-	ar -rs liborac.a $(ORAC_OBJ) $(PARSER_OBJ)
+	$(AR) -rs liborac.a $(ORAC_OBJ) $(PARSER_OBJ)
 
 # Rebuild dependencies housed in dependencies.inc
 depend:
