Subject: Help cmake find ecCodes
diff -upN libemos-4.4.9-Source/CMakeLists.txt libemos/CMakeLists.txt
--- libemos-4.4.9-Source/CMakeLists.txt	2017-06-27 17:08:16.000000000 +0100
+++ libemos/CMakeLists.txt	2019-01-31 16:58:46.002147315 +0000
@@ -131,9 +131,9 @@
                     DEFAULT ON )
 
 if( HAVE_ECCODES)
-  set( GRIB_API_INCLUDE_DIRS ${ECCODES_INCLUDE_DIRS} )
-  set( GRIB_API_LIBRARIES    ${ECCODES_LIBRARIES} )
-  set( GRIB_API_DEFINITIONS  ${ECCODES_DEFINITIONS} )
+  set( GRIB_API_INCLUDE_DIRS ${eccodes_BASE_DIR}/include )
+  set( GRIB_API_LIBRARIES    "eccodes;eccodes_f90;m;jpeg" )
+  set( GRIB_API_DEFINITIONS  "" )
   set( grib_api_BASE_DIR     ${eccodes_BASE_DIR} )
   set( grib_handling_pkg eccodes )
 else()
--- libemos-4.4.9-Source/tests/CMakeLists.txt	2024-08-01 11:16:57.408091984 +0100
+++ libemos/tests/CMakeLists.txt	2024-08-01 11:18:28.251940969 +0100
@@ -161,7 +161,7 @@ if( HAVE_LIBEMOS_GRIBEX )
     ecbuild_add_test( TARGET gribex_demo  SOURCES gribex_demo.F  LIBS emos ENVIRONMENT ${_emos_environment}
       ARGS -i "${PROJECT_SOURCE_DIR}/tests/data/msl_regular_ll.grib" )
 
-    set_tests_properties( gribex_abort PROPERTIES WILL_FAIL ${HAVE_GRIBEX_ABORT} )
+    set_tests_properties( gribex_abort PROPERTIES WILL_FAIL 1 )
     set_tests_properties( gribex_demo  PROPERTIES WILL_FAIL ${HAVE_GRIBEX_ABORT} )
 
 endif( HAVE_LIBEMOS_GRIBEX )
