Subject: Switch to conda build environment
--- a/make.inc	2022-07-06 16:06:38.254703392 +0100
+++ b/make.inc	2022-07-06 16:07:14.326134640 +0100
@@ -1,13 +1,13 @@
 # C compiler and C compiler flags
 
-#CC      = $(CC)
-CCFLAGS = $(CFLAGS) -O3 -cpp -I. -fopenmp
+CC      = $(GCC)
+CCFLAGS = $(CFLAGS) -cpp -I.
 
 # Fortran compiler and Fortan compiler flags (required for the Fortran
 # interface)
-F90      = $(FC)
-F90FLAGS = $(FFLAGS) -O3 -I. -ffree-line-length-0 -cpp -fcheck=bounds -fopenmp
+F90      = $(GFORTRAN)
+F90FLAGS = $(FFLAGS) -I. -ffree-line-length-0 -cpp -fcheck=bounds
 
-LINKS = $(LDFLAGS) -lm -fopenmp
+LINKS = $(LDFLAGS) -lm
 
 # Include and lib directories for non standard locations required by hsd_reader
